payu/models/fms.py

```diff
@@ -65,9 +66,20 @@ def collate(self):
         # Set the stacksize to be unlimited
         res.setrlimit(res.RLIMIT_STACK, (res.RLIM_INFINITY, res.RLIM_INFINITY))
 
+        collate_conf = self.expt.config.get('collate',{})
+
+        if type(collate_conf) is bool:
+            # Cycle through old collate config and convert to newer dict format
+            collate_conf = {'enable' : collate_conf}
+            for key in ['exe','flags','ignore']:
+                oldkey = "collate_{}".format(key)
+                print("Use of this key is deprecated: {}. Use collate dictionary and subkey {}".format(oldkey,key))
+                if oldkey in self.expt.config:
+                    collate_conf[key] = self.expt.config[oldkey]
+
```

Comment:
Similarly, I think it might be possible to convert these keys inside `read_config`?

Reply:
Also may be useful to do a `pop` instead of a read from `config`, then replace `config['collate']` with the updated dict.
Though this could potentially cause problems elsewhere.

Reply:
Also, it's probably OK to just transform everything starting with `collate_`, unless we are just being extra-safe here about our legacy namespace.

Reply:
I guess I was only doing the ones that I were explicitly being used in the FMS model. The other ones are to do with PBS submission. I don't mind either, but the way I'm doing it here it is explicit about what is being used. Adding a new flag won't be affected, as it won't have a legacy equivalent.

Reply:
I agree it would be nice to make all these transformations in a single place, which is what you're suggesting in other comments.